This really disables altivec if not wanted.
Patch from xtophe: https://trac.videolan.org/vlc/ticket/1055
https://bugs.gentoo.org/show_bug.cgi?id=158126
Causes a compile failure because altivec.h is found and -maltivec not appended to cflags
updated to try to fix this

Merged from previous PiSi vlc altivec patch

--- configure.ac.orig	2009-07-09 15:31:14.782154328 +0300
+++ configure.ac	2009-07-09 15:34:37.376154634 +0300
@@ -1570,10 +1570,13 @@
     CPPFLAGS="${CPPFLAGS_save} ${ac_cv_c_altivec}"
   ])
 
-  AC_CHECK_HEADERS(altivec.h)
+  AS_IF([test "${enable_altivec}" != "no"], [
+    AC_CHECK_HEADERS(altivec.h)
+  ])
+
   CPPFLAGS="${CPPFLAGS_save}"
 
-  AS_IF([test "${ac_cv_c_altivec}" != "no"], [
+  AS_IF([test "${ac_cv_c_altivec}" != "no" -a "${enable_altivec}" != "no"], [
     AC_DEFINE(CAN_COMPILE_C_ALTIVEC, 1,
               [Define to 1 if C AltiVec extensions are available.])
     VLC_ADD_CFLAGS([libvlccore],[${ac_cv_c_altivec}])
